pkg/virtctl/imageupload/imageupload.go

```diff
@@ -545,6 +541,54 @@ func createUploadDataVolume(client kubecli.KubevirtClient, namespace, name, size
 	return dv, nil
 }
 
+func createStorageSpec(client kubecli.KubevirtClient, size, storageClass, accessMode string, blockVolume bool) (*cdiv1.StorageSpec, error) {
+	quantity, err := resource.ParseQuantity(size)
+	if err != nil {
+		return nil, fmt.Errorf("validation failed for size=%s: %s", size, err)
+	}
+
+	spec := &cdiv1.StorageSpec{
+		Resources: v1.ResourceRequirements{
+			Requests: v1.ResourceList{
+				v1.ResourceStorage: quantity,
+			},
+		},
+	}
+
+	if storageClass == "" {
+		storageClass, err = getDefaultStorageClassName(client)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if storageClass != "" {
+		spec.StorageClassName = &storageClass
+	}
+
+	if accessMode == "" {
+		accessMode, err = getRecommendedAccessModeFromProfile(client, storageClass)
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
is there a reason why CDI isn't handling the defaulting of accessMode and storageClass based on the cdi config recommendations on the CDI backend? meaning in the DataVolume controller?

Reply:
good catch, the whole point of the PR is to use dv.spec.storage so CDI should handle defaults
I forgot to change this from the previous attempt on another PR. Now corrected.

And the only reason why we read defaults is to provide a fast feedback to the user. Do not create a DV that we know can't succeed.